swiftpkg/internal/swiftpkg_build_files.bzl

```diff
@@ -106,7 +106,7 @@ def _swift_target_build_file(pkg_ctx, target):
     if len(defines) > 0:
         attrs["defines"] = bzl_selects.to_starlark(defines)
     if len(copts) > 0:
-        attrs["copts"] = bzl_selects.to_starlark(copts)
+        attrs["copts"] = bzl_selects.unwrap_list(copts)
```

Comment:
Is there any way to detect this in `bzl_selects.to_starlark()`? It is not obvious to me why `copts` would get this special handling.

Reply:
The reason is that we add a string repeatedly:
https://github.com/cgrindel/rules_swift_package_manager/blob/d671d9e8b5f2fc7e5c65eb9bbe072cf62d04d387/swiftpkg/internal/swiftpkg_build_files.bzl#L103

`to_starlark` relies on using sets within its processing, which will deduplicate the string that's added.

I'm not familiar with the repository, so I decided to add a new function to reduce the chances that I'll affect other places that use `to_starlark`.

Possible Alternatives:
----------------------
- Add a parameter like `dedup = True`, and only perform deduplication after the list has been created, so that if `dedup = False` was passed, it will return the list without deduping.
- Move adding `-enable-experimental-feature` after running `to_starlark`, so that it skips the deduplication. This allows us to reuse `to_starlark` as it is now. Experimental features will need to be placed in a different list, so it doesn't affect the other `copts` from unsafe flags.

Reply:
I tried doing `-enable-experimental-feature` after running `to_starlark`, but didn't realise that it returns a structure instead of a list. So this method won't work, as `to_starlark` will have copts in an internal structure. (I did not dig into whether it'll be easy to append values into that list).

I'll give adding the dedup parameter a try (maybe renamed to `preserve_list`?)